gobfile.go

```diff
@@ -16,8 +16,8 @@ func readGobFile(fsys fs.FileSystem, name string, v interface{}) error {
 	return dec.Decode(v)
 }
 
-func writeGobFile(fsys fs.FileSystem, name string, v interface{}) error {
-	f, err := openFile(fsys, name, true)
+func writeGobFile(fsys fs.FileSystem, name string, v interface{}, readOnly bool) error {
```

Comment:
What would be the desired effect of writing to a file opened in read-only mode? Wouldn't it always fail?